xbmc/addons/kodi-dev-kit/include/kodi/Filesystem.h

```diff
@@ -2017,24 +2017,51 @@ class ATTR_DLL_LOCAL CFile
   /// @param[out] line The buffer to store the data in.
   /// @return True when a line was read, false otherwise.
   ///
+  ///
+  /// ------------------------------------------------------------------------
+  ///
+  /// **Example:**
+  /// ~~~~~~~~~~~~~{.cpp}
+  /// #include <kodi/Filesystem.h>
+  ///
+  /// ...
+  ///
+  /// /* Create the needed file handle class */
+  /// kodi::vfs::CFile myFile;
+  ///
+  /// /* Open the wanted file */
+  /// if (myFile.OpenFile(kodi::addon::GetUserPath("/myFile.txt")))
+  /// {
+  ///   /* Read all lines inside file */
+  ///   while (1)
+  ///   {
+  ///     std::string line;
+  ///     if (!myFile.ReadLine(line))
+  ///       break;
+  ///     fprintf(stderr, "%s\n", line.c_str());
+  ///   }
+  /// }
+  ///
+  /// ~~~~~~~~~~~~~
+  ///
   bool ReadLine(std::string& line)
   {
     using namespace kodi::addon;
 
     line.clear();
     if (!m_file)
       return false;
-    // TODO: Read 1024 chars into buffer. If file position advanced that many
-    // chars, we didn't hit a newline. Otherwise, if file position is 1 or 2
-    // past the number of bytes read, we read (and skipped) a newline sequence.
-    char buffer[1025];
-    if (CPrivateBase::m_interface->toKodi->kodi_filesystem->read_file_string(
-            CPrivateBase::m_interface->toKodi->kodiBase, m_file, buffer, sizeof(buffer)))
```

Comment:
@garbear ignoring the new code, could the fix be as simple as having the buffer size be `-1` in the old code. So:

```cpp
 char buffer[1025];
    if (CPrivateBase::m_interface->toKodi->kodi_filesystem->read_file_string(
            CPrivateBase::m_interface->toKodi->kodiBase, m_file, buffer, sizeof(buffer -1)))
    {
      line = buffer;
      return !line.empty();
    }
    return false;
```

Reply:
`sizeof(buffer -1)` makes no sense. You probably meant `sizeof(buffer)-1`

Reply:
Correct, will fix shortly.